versioned/tiered/rocksdb/src/main/java/org/projectnessie/versioned/rocksdb/RocksDBStore.java

```diff
@@ -0,0 +1,386 @@
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import org.projectnessie.versioned.impl.condition.ConditionExpression;
+import org.projectnessie.versioned.impl.condition.UpdateExpression;
+import org.projectnessie.versioned.store.ConditionFailedException;
+import org.projectnessie.versioned.store.Id;
+import org.projectnessie.versioned.store.LoadOp;
+import org.projectnessie.versioned.store.LoadStep;
+import org.projectnessie.versioned.store.NotFoundException;
+import org.projectnessie.versioned.store.SaveOp;
+import org.projectnessie.versioned.store.Store;
+import org.projectnessie.versioned.store.ValueType;
+import org.projectnessie.versioned.tiered.BaseValue;
+import org.rocksdb.ColumnFamilyDescriptor;
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.ColumnFamilyOptions;
+import org.rocksdb.DBOptions;
+import org.rocksdb.FlushOptions;
+import org.rocksdb.Options;
+import org.rocksdb.ReadOptions;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.RocksIterator;
+import org.rocksdb.Transaction;
+import org.rocksdb.TransactionDB;
+import org.rocksdb.TransactionDBOptions;
+import org.rocksdb.WriteBatch;
+import org.rocksdb.WriteOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.AbstractIterator;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * A RocksDB specific backing store to store and retrieve Nessie metadata.
+ */
+public class RocksDBStore implements Store {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(RocksDBStore.class);
+  private static final RocksDBValueVisitor VALUE_VISITOR = new RocksDBValueVisitor();
+  private static final String DEFAULT_COLUMN_FAMILY = new String(RocksDB.DEFAULT_COLUMN_FAMILY, UTF_8);
+
+  static {
+    RocksDB.loadLibrary();
+  }
+
+  private TransactionDB rocksDB;
+  private Map<ValueType<?>, ColumnFamilyHandle> valueTypeToColumnFamily;
+  private final RocksDBStoreConfig config;
+
+  /**
+   * Creates a store ready for connection to RocksDB.
+   * @param config the configuration for the store.
+   */
+  public RocksDBStore(RocksDBStoreConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void start() {
+    final String dbPath = verifyPath();
+    final List<ColumnFamilyDescriptor> columnFamilies = getColumnFamilies(dbPath);
+
+    try (final DBOptions dbOptions = new DBOptions().setCreateIfMissing(true).setCreateMissingColumnFamilies(true)) {
+      // TODO: Consider setting WAL limits.
+      final List<ColumnFamilyHandle> columnFamilyHandles = new ArrayList<>();
+      rocksDB = TransactionDB.open(dbOptions, new TransactionDBOptions(), dbPath, columnFamilies, columnFamilyHandles);
+      final ImmutableMap.Builder<ValueType<?>, ColumnFamilyHandle> builder = new ImmutableMap.Builder<>();
```

Comment:
It would be nice if there's a blank line to separate conceptual blocks of functionality.

Reply:
Blank line added.